vendor/engines/split_accounts/spec/factories/split_accounts.rb

```diff
@@ -0,0 +1,45 @@
+FactoryGirl.define do
+  factory :split_account do
+
+    transient do
+      owner { create(:user) }
+      without_splits { false } # set to true to skip generating a valid split
+    end
+
+    sequence(:account_number) { |n| "account_number_#{n}" }
+    sequence(:description) { |n| "split account #{n}" }
+    expires_at { Time.zone.now + 1.month }
+    created_by 0
+
+    # Every account must have an account_user "owner" in order for the account
+    # to be valid in rails. And foreign key constraints require that each
+    # account_user has an account inserted before the account_user is inserted.
+    callback(:after_build, :before_create) do |split_account, evaluator|
+      split_account.account_users << build(:account_user, user: evaluator.owner, account: split_account)
+    end
+
+    trait :with_two_splits do
+      callback(:after_build, :before_create) do |split_account, evalutor|
+        split_account.splits << build(:split, percent: 50, extra_penny: true, parent_split_account: split_account)
+        split_account.splits << build(:split, percent: 50, extra_penny: false, parent_split_account: split_account)
+      end
+    end
+
+    trait :with_three_splits do
+      callback(:after_build, :before_create) do |split_account, evalutor|
+        split_account.splits << build(:split, percent: 33.33, extra_penny: true, parent_split_account: split_account)
+        split_account.splits << build(:split, percent: 33.33, extra_penny: false, parent_split_account: split_account)
+        split_account.splits << build(:split, percent: 33.34, extra_penny: false, parent_split_account: split_account)
+      end
+    end
+
+    # Leave this at the bottom of the factory.
+    # Add a split if none exist and if transient `without_splits` is false.
+    callback(:after_build, :before_create) do |split_account, evaluator|
```

Comment:
I think so, if we ever do some ajax-y UI stuff, allowing this will come in handy.